src/main/java/org/opentest4j/AssertionFailedError.java

```diff
@@ -34,72 +32,50 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final Serializable expected;
-	private final Serializable actual;
+	private final ValueWrapper expected;
+	private final ValueWrapper actual;
 
 	public AssertionFailedError() {
 		this(null);
 	}
 
 	public AssertionFailedError(String message) {
-		this(message, Undefined.INSTANCE, Undefined.INSTANCE);
+		this(message, null);
 	}
 
-	public AssertionFailedError(String message, Serializable expected, Serializable actual) {
+	public AssertionFailedError(String message, Object expected, Object actual) {
 		this(message, expected, actual, null);
 	}
 
 	public AssertionFailedError(String message, Throwable cause) {
-		this(message, Undefined.INSTANCE, Undefined.INSTANCE, cause);
+		this(message, (ValueWrapper) null, (ValueWrapper) null, cause);
```

Comment:
This seems to be intended to select the right constructor call. Under JDK 1.7 and 1.8 it seems to properly select the private constructor even without the cast. Did you have a failing test without the cast?